OpenNMS/horizon-stream
# HS-581 : Generate new event from suspect traps and sent to internal event topic

## Description

What changed: Create a new event with a suspect Trap message and send it to a new internal event topic. Consume this new event in the inventory and create a new node if the IP interface with the same location and tenant id doesn't exist in the database. 

How to test:
- Wire up the environment with tilt or other k8s tools
- Create a snmp pod with command: _kubectl run snmpd-node --image polinux/snmpd --restart=Never_ 
- Get the minion pod IP address
- Run this command to get into the new created snmp pod _kubectl exec -it pod/snmpd-node -- /bin/sh_
- Run this command inside the snmp pod with minion IP address: _/usr/bin/snmptrap -v 2c -c public <minion IP address>:1162 '' 1.3.6.1.4.1.8072.2.3.0.1 1.3.6.1.4.1.8072.2.3.2.1 i 123456_
- Wait for a few minutes and check the new node created in the Inventory tab on the UI page.

## Jira link(s)
- https://issues.opennms.org/browse/HS-581
- https://issues.opennms.org/browse/HS-653
- https://issues.opennms.org/browse/HS-654

## Flagged for review


## Checklist
* [x] Follows Horizon Stream's [development guidelines.](https://github.com/OpenNMS/horizon-stream/wiki/Development-Guidelines)
* [x] Appropriate reviewer(s) have been selected.
* [x] Jira issue(s) have been updated to "In Review".
* [x] Includes [appropriate tests.](https://github.com/OpenNMS/horizon-stream/wiki/Test-Strategy)
